web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -334,8 +346,17 @@
     }
     $isShowDetail = !$isShowDetail;
   };
+  const showEditorHandler = () => {
+    if (isShowActivity) {
+      isShowActivity = false;
+    }
+    isShowEditor = !isShowEditor;
+  };
 
   const trashOrDelete = async (force: boolean = false) => {
+    if (!showNavbar) {
```

Comment:
If navbar isn't shown, can you actually click on these button?

Reply:
no, navbar is hidden when the editor is open. and their shortcuts are disabled. only while the editor is open
![image](https://github.com/user-attachments/assets/58d2f4ff-0b43-4e79-ad71-8b6a61ba9eed)

Reply:
Then we don't really need the logic here right? Because there is no way this action is triggered if all the buttons and shortcut are hidden/disabled

Reply:
```js
const trashOrDelete = async (force: boolean = false) => {
  if (!showNavbar) {
```
it is needed to prevent a call through a shortcut. the `trashOrDelete` function is called in the shortcut
```js
{ shortcut: { key: 'Delete' }, onShortcut: () => trashOrDelete(asset.isTrashed) },
{ shortcut: { key: 'Delete', shift: true }, onShortcut: () => trashOrDelete(true) },
```

Reply:
This is debatable, but do you think it would be more precise to handle it in the `onShortcut` action? :thinking: 

My thinking is that we keep the `trashOrDelete` and other actions to just do what it is meant to do when it gets invoked.

Reply:
Yes, I also had doubts about the right place to do it.
But this method, in my opinion, looks very cumbersome:
```js
<svelte:window
  use:shortcuts={[
    { shortcut: { key: 'a', shift: true }, onShortcut: ()=>showNavbar?toggleAssetArchive:null },
    { shortcut: { key: 'ArrowLeft' }, onShortcut: () => showNavbar?navigateAsset('previous'):null },
    { shortcut: { key: 'ArrowRight' }, onShortcut: () => showNavbar?navigateAsset('next'):null },
    { shortcut: { key: 'd', shift: true }, onShortcut: () => (showNavbar ? downloadFile(asset) : null) },
    { shortcut: { key: 'Delete' }, onShortcut: () => showNavbar?trashOrDelete(asset.isTrashed):null },
    { shortcut: { key: 'Delete', shift: true }, onShortcut: () => showNavbar?trashOrDelete(true):null },
```

Should I make a wrapper?
```js
function handleShortcut(condition, action) => {
    return () => (condition ? action() : null);
  };
```

Reply:
#11449 should improve this situation, I think you can also do `onShortcut: () => showNavbar && toggleAssetArchive` to keep is a bit shorter, but both aren't an ideal solution

Reply:
done